modules/internal/SharedObject.chpl

```diff
@@ -19,40 +19,81 @@
 
 /*
 
-   :record:`Shared` (along with :record:`~OwnedObject.Owned`) manage the
-   deallocation of a class instance. :record:`Shared` is meant to be used when
-   many different references will exist to the object and these references need
-   to keep the object alive.
+:record:`shared` (along with :record:`~OwnedObject.owned`) manage the
+deallocation of a class instance. :record:`shared` is meant to be used when
+many different references will exist to the object and these references need
+to keep the object alive.
 
-   To use :record:`Shared`, allocate a class instance following this
-   pattern:
+Using `shared`
+--------------
 
-   .. code-block:: chapel
+To use :record:`shared`, allocate a class instance following this
+pattern:
 
-     var mySharedObject = new Shared(new MyClass(...));
+.. code-block:: chapel
 
-   When mySharedObject and any copies of it go out of scope, the class
-   instance it refers to will be deleted.
+ var mySharedObject = new shared MyClass(...));
 
-   Copy initializing or assigning from mySharedObject will make
-   other variables refer to the same class instance. The class instance
-   will be deleted after all of these references go out of scope.
+When ``mySharedObject`` and any copies of it go out of scope, the class
+instance it refers to will be deleted.
 
-   .. code-block:: chapel
+Copy initializing or assigning from mySharedObject will make
+other variables refer to the same class instance. The class instance
+will be deleted after all of these references go out of scope.
 
-     var globalSharedObject:Shared(MyClass);
+.. code-block:: chapel
 
-     proc makeGlobalSharedObject() {
-       var mySharedObject = new Shared(new MyClass(...));
-       globalSharedObject = mySharedObject;
-       // now mySharedObject is deinitialized, but the MyClass
-       // instance is not deleted until globalSharedObject is deinitialized.
-     }
+ var globalSharedObject:shared MyClass;
 
-   :record:`Shared` supports coercions to the class type as well as
-   coercions from a ``Shared(T)`` to ``Shared(U)`` where ``T`` is a
-   subclass of ``U``. See :record:`~OwnedObject.Owned` for examples
-   of these coercions.
+ proc makeGlobalSharedObject() {
+   var mySharedObject = new shared new MyClass(...);
+   globalSharedObject = mySharedObject;
+   // now mySharedObject is deinitialized, but the MyClass
```

Comment:
yes (assuming you mean reference count decreased). mySharedObject is a record though, and that record is deinitialized. I'll adjust it, since the record-ness is probably a distraction here.

Reply:
I was confused because I thought the comment applied to the assignment at line 50 vs. going out of scope at line 55.  I think the updated version is more clear. Thanks.